src/tools/run-make-support/src/lib.rs

```diff
@@ -401,12 +401,26 @@ pub fn recursive_diff(dir1: impl AsRef<Path>, dir2: impl AsRef<Path>) {
     });
 }
 
-pub fn read_dir<F: FnMut(&Path)>(dir: impl AsRef<Path>, mut callback: F) {
+pub fn read_dir<F: FnMut(&Path), P: AsRef<Path>>(dir: P, mut callback: F) {
     for entry in fs_wrapper::read_dir(dir) {
         callback(&entry.unwrap().path());
     }
 }
 
+pub fn get_files_with_extension<P: AsRef<Path>, S: AsRef<OsStr>>(
```

Comment:
We already have the extension helpers and a more general `get_files` helper here: https://github.com/rust-lang/rust/pull/126957/files#diff-7300c8aabb7002edfa4b02c399458c2bcc2c71674dad268f9ea2fd5472adb37eR252. I still think that `get_files_with_extension` is common enough to be useful here, but we should base this on top of that PR, since the other helper is more general.

Reply:
Since the other PR is blocked, what should I do here?

Reply:
Well, it is blocked, because this function has been introduced in like 4 parallel PRs :laughing: We just need to wait until they are all merged in succession, I don't think it should take more than a few days. There were about 20 tests ported in 5 PRs that needed similar functionality, so we need to serialize their merging to unwrap all of this.